text/0000-compiler-team-trusted-contrib-maintainer.md

```diff
@@ -0,0 +1,401 @@
+]
+
+[[roles]]
+id = "compiler-backport"
+description = "Backport Reviewer"
+```
+
+By tracking responsibility participation in the team repository, it is easier
+for the team leads to have visibility into the participation in each to ensure
+that it is sustainable. Participants will be recognised for the additional
+responsibilities they participate in on the compiler team's page on the Rust
+website.
+
+Drawbacks
+=========
+[drawbacks]: #drawbacks
+
+- Granting permissions earlier may be a risk
+    - We haven't had any issues with contributors having staying power to the
+      extent that we would trust them with permissions and then having those be used
+      inappropriately. We can always revert changes if necessary.
+- Expectations of team members
+    - This RFC formally establishes expectations which come with team
+      membership. Some team members already assume that these expectations are there,
+      but this wasn't made explicit when current team members were made team members.
+
+Rationale and alternatives
+==========================
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- Get better at doing nominations
+    - A lot of this proposal's simplification of the way that promotions are
+      granted is based on the premise that our current system doesn't work well for us
+      - but we could just try and do the current system better.
+- Only change review responsibilities
+    - We could instead try to increase the number of reviewers on the review
+      queue by just amending the current compiler team membership policy to include
+      review rotation duty. This does not improve our ability to correctly promote
+      contributors or recognize the ways individuals contribute to the maintenance
+      of the compiler but could be reasonable if implementing all of the changes
+      described here will take too long.
+
+Prior art
+=========
+[prior-art]: #prior-art
+
+- [Responsibilities][responsibilities] are similar to [an unsubmitted proposal
+  by Niko Matsakis in December 2020 to have "elected officers"][officers] within
+  the compiler team responsible for different team functions. This RFC shares
+  many of the goals of Niko's earlier proposal, but is slightly less formal -
+  responsibilities are loosely-defined groups of contributors rather than elected
+  positions, and there is no rotations or term limits.
+
+  In this RFC's proposal, it is expected that responsibilities are shared
+  amongst a group of team members, and that team members do less of other
+  responsibilities so that their workload is sustainable, but this isn't enforced.
+  Team leads are instead responsible for ensuring that the team is large enough to
+  perform each responsibility sustainably.
+
+[officers]: https://hackmd.io/S9xqmwJbSI-a9mFdK9yQBA
+
+Unresolved questions
+====================
+[unresolved-questions]: #unresolved-questions
+
+None!
+
+Future possibilities
```

Comment:
I think another future possibility is further elaboration of some minimum requirement for maintainers.

Reply:
Yes, I was also thinking to something along that line esp. regarding being a more stable presence in the Rust project (see [comment](https://github.com/rust-lang/rfcs/pull/3599/files#r1551099380)).

After reading this document, I infer that maintainers of the Rust compiler are expected to contribute more consistently than team members.

Reply:
Added some wording for this.